content/post/new-python-api.md

```diff
@@ -0,0 +1,263 @@
+* Source -- Input point for data from external systems into an application.
+* Sink -- Output point from an application to external systems.
+* Decoder -- Code that transforms a stream of bytes from an external
+  system into a series of application input types.
+* Encoder -- Code that transforms an application output type into
+  bytes for sending to an external system.
+* Pipeline -- A sequence of computations and/or state computations
+  originating from a source and optionally terminating in a sink.
+* Application -- A collection of pipelines.
+
+We’ll cover these abstractions in more detail as we proceed.
+
+### Application Setup
+
+Wallaroo calls the `application_setup` function to create a data structure that represents the application.
+
+```
+def application_setup(args):
+    in_host, in_port = wallaroo.tcp_parse_input_addrs(args)[0]
+    out_host, out_port = wallaroo.tcp_parse_output_addrs(args)[0]
+
+    word_partitions = list(string.ascii_lowercase)
+    word_partitions.append("!")
+
+    ab = wallaroo.ApplicationBuilder("Word Count Application")
+    ab.new_pipeline("Split and Count",
+                    wallaroo.TCPSourceConfig(in_host, in_port, decoder))
+    ab.to_parallel(split)
+    ab.to_state_partition(count_word, WordTotals, "word totals",
+        partition, word_partitions)
+    ab.to_sink(wallaroo.TCPSinkConfig(out_host, out_port, encoder))
+    return ab.build()
+```
+
+This code creates an application with the topology that was described earlier. It represents one pipeline that consists of a stateless computation called `Split` that splits a string of words into individual words and a state computation called `CountWord` that updates the state of the application and creates outgoing messages that represent the word count. The objects and functions used here will be described more in the following sections.
+
+#### State Partitions and State Entities
+
+In this example, the state is the number of times each word has been seen. The easiest way to do this would be with a dictionary where the key is a word, and the value associated with that key is the number of times that word has been seen in the event stream.
+
+Wallaroo lets you divide state partitions into pieces called state entities. State entities are pieces of state partitions that are uniquely identified by a key of some sort. State can be partitioned by any number of keys. The only restriction is that the state entities must be completely isolated from each other so that they can be accessed and updated independently.
+
+When a message is sent, Wallaroo applies a partition function to the message to determine which state partition to send it to. Different state entities may live on different workers, and an entity may move from one worker to another when workers are added or removed from the cluster. This makes it easy to scale the application up and down as the number of workers in the cluster increases and decreases.
```

Comment:
There seems to be a jump between `state partition` and `state entity` here. The first sentence says that the `partition function to the message to determine which state partition to send it to` but then the next sentence starts talking about `state entities` living on different workers. I know the previous paragraph defined that the `state entities` are subsets of the `state partitions` but this still feels off to me.